src/main/java/org/folio/rest/impl/InventoryHelper.java

```diff
@@ -626,4 +646,60 @@ private JsonObject getFirstObjectFromResponse(JsonObject response, String proper
                    .map(item -> (JsonObject) item)
                    .orElseThrow(() -> new CompletionException(new InventoryException(String.format("No records of '%s' can be found", propertyName))));
   }
+
+  /**
+   * Returns JsonObject with id of ContributorNameType retrieved from inventory-storage.
+   *
+   * @param contributorNameTypeName ContributorNameType
+   * @return JsonObject ContributorNameTypeId
+   */
+  private CompletableFuture<JsonObject> getContributorNameTypeId(ContributorNameTypeName contributorNameTypeName) {
+    UnaryOperator<String> endpoint = query -> String.format("/contributor-name-types?query=name==%s", encodeQuery(query, logger));
+    return cacheAndGet(contributorNameTypeName.getName(),
+      endpoint,
+      entries -> {
+        JsonObject contributorPersonalNameType = new JsonObject();
+        contributorPersonalNameType.put(CONTRIBUTOR_NAME_TYPE_ID, getFirstObjectFromResponse(entries, CONTRIBUTOR_NAME_TYPES).getString(ID));
+        return contributorPersonalNameType;
+      });
+  }
+
+  /**
+   * Caches value in Vert.X Context and returns it by key.
+   *
+   * @param key key for retrieving value from cache
+   * @param endpointConstructor endpoint for GET request to retrieve value
+   * @param fn Function<JsonObject, JsonObject> for transformation JsonObject to new one if needed
+   *
+   * @return value from cache
+   */
+  private CompletableFuture<JsonObject> cacheAndGet(String key, UnaryOperator<String> endpointConstructor, Function<JsonObject, JsonObject> fn) {
```

Comment:
I would think "getAndCache" might be a better name, but it's certainly not a big deal, especially given simplicity of the function.